auth/auth.ts

```diff
@@ -105,19 +106,25 @@ export class MockAuth implements firebase.auth.Auth {
 
   private signIn(
     user: User,
-    additionalUserInfo: firebase.auth.AdditionalUserInfo | null = null
+    { isNewUser }: { isNewUser: boolean }
   ): Promise<firebase.auth.UserCredential> {
     this.currentUser = user;
     this.authStateEvents.forEach((listener) => {
       listener(user);
     });
 
-    return Promise.resolve<firebase.auth.UserCredential>({
-      user,
-      additionalUserInfo,
-      credential: null,
-      operationType: "signIn",
-    });
+    return Promise.resolve(new UserCredential(user, "signIn", isNewUser));
+  }
+
+  private async signInWithSocial(provider: firebase.auth.AuthProvider) {
```

Comment:
Maybe we use `@firebase/auth-types` to avoid having to use `firebase.auth` for every one of these types.

Reply:
> This package is not intended for direct usage, and should only be used via the officially supported firebase package.

😕 from https://www.npmjs.com/package/@firebase/auth-types/v/0.10.1

Reply:
Yes, but that is simply because it is intended to be a private API for firebase, we are already very much working with internal APIs when mocking.